### PR TITLE
Upgrade to Om 0.8.7

### PR DESCRIPTION
**Pull request by sritchie**
React components clone great now, but Om components are still hosed.

### TODO
- [x] Fix for clone-with-props in react 0.12.2

### CODE REVIEW

**Comment by sritchie** [comment]
Thanks to @swannodette for this tip: https://github.com/swannodette/om/commit/ca93b5f531399ca407a2766c88e7a3db41e47f46

Should be helpful in cleaning up the current code.

**Comment by dignati** [comment]
@sritchie Any progress on this? Let me know if I can help!

**Comment by sritchie** [comment]
Note for @sritchie - Check out the deprecation warnings here: http://facebook.github.io/react/blog/2014/10/28/react-v0.12.html

**Comment by sritchie** [comment]
Okay, I think this is fixed! @dignati, I'm deploying 0.4.0-SNAPSHOT now. Can you give it a try and see if you're good? Then we can merge this and I can do a full release.

File: docs/src/clj/om_bootstrap/server.clj [hunk 1]
```diff
@@ -20,7 +20,7 @@
 (defn clojurescript []
   (if (dev?)
     (list
-     (h/include-js "http://fb.me/react-0.11.1.js")
+     (h/include-js "http://fb.me/react-0.12.2.js")
```
Comment (sritchie) [inline docs/src/clj/om_bootstrap/server.clj]: We really should just add this to the preamble.

Comment (dignati) [inline docs/src/clj/om_bootstrap/server.clj]: @sritchie Is this even needed? Om includes it anyway I think.

**Comment by dignati** [comment]
The docs page looks good to me except this warning: 
```
Warning: createElement(...): Encountered component with a `key` of null. In a future version, this will be treated as equivalent to the string 'null'; instead, provide an explicit key or use undefined.
```

I could test a little more on Monday and try it on a real project.

File: docs/src/clj/om_bootstrap/server.clj (hunk 1 above)
Comment (sritchie) [inline docs/src/clj/om_bootstrap/server.clj]: It's needed unless you explicitly add the React file to the :preamble in cljsbuild. Making that change now.

**Comment by sritchie** [comment]
@dignati, fixed that issue. Looks like the tests failed - looking at that too. I'll push a new SNAPSHOT without the warnings.

**Comment by sritchie** [comment]
@dignati snapshot's up. LMK when you get a chance to try it. I'll merge and release monday otherwise.

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Here's the latest Leiningen version info:
 You'll also need to add Om:
 
 ```clojure
-[om "0.7.3"]
+[org.omcljs/om "0.8.7"]
```
Comment (dpetrovics) [inline README.md]: nice

**Comment by dpetrovics** [comment]
@sritchie looks good to me, merge when ready

**Pull request merged**